libp2p/switch.nim

```diff
@@ -189,41 +192,40 @@ proc upgradeIncoming(s: Switch, conn: Connection) {.async, gcsafe.} =
   # handle secured connections
   await ms.handle(conn)
 
-proc dial*(s: Switch, 
-           peer: PeerInfo, 
-           proto: string = ""): 
-           Future[Connection] {.async.} = 
+proc dial*(s: Switch,
+           peer: PeerInfo,
+           proto: string = ""):
+           Future[Connection] {.async.} =
   let id = peer.id
-  trace "dialing peer", peer = id
-  for t in s.transports: # for each transport
-    for a in peer.addrs: # for each address
-      if t.handles(a): # check if it can dial it
-        if id notin s.connections:
-          trace "dialing address", address = $a
+  trace "Dialing peer", peer = id
+  result = s.connections.getOrDefault(id)
+  if result == nil or result.closed:
```

Comment:
Just a general question, whats the advantage of `== nil` over `isNil()`?

Reply:
There is not much of a difference, it's mostly about habits. You could argue that `isNil` is slightly safer, because sometimes pointer-like types tend to override `==` which creates surprising effects with `x == nil`.